plugins/woocommerce-blocks/assets/js/blocks/product-collection/deprecated.tsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import { BlockInstance, createBlock } from '@wordpress/blocks';
+import { useInnerBlocksProps, useBlockProps } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { LayoutOptions, ProductCollectionLayout } from './types';
+
+interface DisplayLayoutAttribute {
+	type: 'list' | 'flex';
+	columns: number;
+	shrinkColumns: boolean;
+}
+
+const migrateDisplayLayout = (
+	attributes: { displayLayout: DisplayLayoutAttribute },
+	innerBlocks: BlockInstance[]
+) => {
+	const { displayLayout = null, ...newAttributes } = attributes;
```

Comment:
```suggestion
	const { displayLayout, ...newAttributes } = attributes;
```

Nitpick: Default value will be assigned only if `displayLayout` is `undefined` and that will be caught in line 23 anyway so I think it's simpler without defaulting to `null`.